src/Features/Core/Portable/MetadataAsSource/VirtualMetadataDocumentPersister.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.MetadataAsSource;
+
+internal sealed class VirtualMetadataDocumentPersister() : IMetadataDocumentPersister
+{
+    public const string VirtualFileScheme = "roslyn-metadata";
+
+    public void CleanupGeneratedDocuments()
+    {
+        // Nothing is ever persisted to disk, so cleanup has no meaning here.
+        return;
+    }
+
+    public string ConvertFilePathToDocumentPath(Guid identifier, string providerName, string filePath)
+    {
+        var fileName = Path.GetFileName(filePath);
+        return GenerateDocumentPath(identifier, providerName, fileName);
+    }
```

Comment:
I'm confused what this method means, especially in the case of the virtual provider here -- where did this filename come from in the first place?

Reply:
Maybe this needs more comments there.